### ISSUE TITLE
wandb changes hydra logging in an undesired manner. [CLI] 

### ISSUE THREAD

**Issue by somedadaism** [wandb/client#2075]
**Description**
I try to combine `hydra` and `wandb` in a common framework.
`hydra` configures my (python, not wandb) loggers in a manner I think is convenient and allows to flexible alter them.
However the logger configurations made by `hydra` are overwritten by the `wandb.init()` function.
The issue exists in public repos too:
https://github.com/ashleve/lightning-hydra-template 
is an example I checked.

**Wandb features**
`wandb.init()`

**How to reproduce**
```
import hydra
import sys
import wandb
import logging
import warnings

log = logging.getLogger(__name__)
warnings.filterwarnings('ignore', category=UserWarning)


@hydra.main(config_path='../configs', config_name='benchmark_config.yaml')
def reproduce_bug(config):
    log.info('foo')
    wandb.init(project='Test',
               dir='..',
               reinit=True)
    log.info('bar')


if __name__ == '__main__':
    reproduce_bug()
"""
output: 
[2021-04-15 11:17:16,642][__main__][INFO] - foo

expected: 
[2021-04-15 11:17:16,642][__main__][INFO] - foo
[2021-04-15 11:17:16,642][__main__][INFO] - bar
```

**Environment**
Multiple, python3.8 on Mac, Ubuntu

I have raised a related issue with additional information on the `hydra` repository:
https://github.com/facebookresearch/hydra/issues/1553

**Comment by ariG23498** [comment]
Hey @somedadaism 
Thanks for the issue and the repro.
We will be tracking this internally. 😄

**Comment by somedadaism** [comment]
Hi there has been activity on the related hydra issue facebookresearch/hydra#1553 which I opened in parallel. It relates this behaviour to the `WANDB_START_METHOD=thread` which is required to allow multirun with hydra.

**Comment by somedadaism** [comment]
Hi I would like to report additional behaviour I observe:
Removing `WANDB_START_METHOD=thread` and hydra multiruns work without issues for me.
**However python logging between `hydra` and W&B remains difficult to harmonize!**
The following example works almost as expected:
```
import hydra
import wandb
import logging
import warnings

log = logging.getLogger(__name__)
warnings.filterwarnings('ignore', category=UserWarning)


@hydra.main(config_path='../configs', config_name='my_config.yaml')
def almost_as_desired(config):
    log.info('foo')
    wandb.init(project='Test',
               dir='..',
               reinit=True)
    print('I like to foobar!')
    log.info('bar')


if __name__ == '__main__':
    almost_as_desired()
```
The terminal output now is - as it should be:
```
[2021-04-17 13:49:48,619][__main__][INFO] - foo
I like to foobar!
[2021-04-17 13:49:50,608][__main__][INFO] - bar
```
The contenst of the hydra log are as expected:
```
[2021-04-17 13:52:32,647][__main__][INFO] - foo
[2021-04-17 13:52:34,677][__main__][INFO] - bar
```
However the wandb run has created the following `output.log` file:
```
[2021-04-17 13:56:29,509][__main__][INFO] - bar
```
This seems ok at first - I don't expect W&B to track logs before it is initialized 
However the message `'I like to foobar!` has gone missing.
I think this is strange/inconsistent as commenting out hydra gives:
```
import hydra
import wandb
import logging
import warnings

log = logging.getLogger(__name__)
warnings.filterwarnings('ignore', category=UserWarning)


# @hydra.main(config_path='../configs', config_name='my_config.yaml')
def reproduce_bug(config=None):
    log.info('foo')
    wandb.init(project='Test',
               dir='..',
               reinit=True)
    print('I like to foobar!')
    log.info('bar')


if __name__ == '__main__':
    reproduce_bug()
```
and the generated `output.log` looks like this:
```
I like to foobar!
```
Of course hydra has changed the verbosity of the `bar` log so we miss it here but shouldn't the printout also be in the version where hydra is used?

**Comment by gwenzek** [comment]
I'm lost here. The fix for this issue has been reverted in #2083 and is sitting in #2085.
What's going on ?

For me wandb is currently totally unusable because of this. 
[Disabling the root handler](https://github.com/wandb/client/pull/2083/files#diff-1383658fa12fa4504e0835b9f3eb487a25e35d0cfe1e064e4dd54021b155ca03R169) is totally unacceptable for a library.
If the root logger has handlers it's because someone created them.

As a consequence none of my job logs are written to the stdout nor to wandb.

I think what you want to do is to add a handler that write to wandb servers, not to disable other handlers.

**Comment by farizrahman4u** [comment]
@gwenzek The first PR was merged at a wrong time during our release cycle so had to be reverted. The merge automatically closed this issue. I am reopening it until the new PR gets merged.
